stylix/mk-target.nix

```
@@ -170,73 +167,164 @@
-        fn:
-        lib.genAttrs (functionArgNames fn) (
-          arg:
-          if arg == "cfg" then
-            cfg
-          else if arg == "colors" then
-            config.lib.stylix.colors
-          else
-            config.stylix.${arg}
-              or (throw "stylix: mkTarget expected one of `cfg`, `colors`, ${
-                lib.concatMapStringsSep ", " (name: "`${name}`") (
-                  builtins.attrNames config.stylix
+      getArguments =
+        function:
+        lib.genAttrs
+          (lib.pipe function [
+            lib.functionArgs
+            builtins.attrNames
+          ])
+          (
+            argument:
+            if argument == "cfg" then
+              cfg
+
+            else
+              (
+                config':
+                let
+                  inherit (cfg.${argument}) override;
+                in
+                if override == null then
+                  config'
+                else if builtins.typeOf override != builtins.typeOf config' then
+                  throw "stylix: expected `config.stylix.targets.${name}.${argument}.override` to be a ${builtins.typeOf config'}, but got: ${builtins.typeOf override})"
+                else if builtins.isAttrs override then
+                  lib.recursiveUpdate config' override
+                else
+                  override
+              )
+                (
+                  if argument == "colors" then
+                    config.lib.stylix.colors
+
+                  else
+                    config.stylix.${argument} or (throw "stylix: mkTarget expected one of ${
+                      lib.concatMapStringsSep ", " (expected: "`${expected}`") (
+                        lib.naturalSort (
+                          [
+                            "cfg"
+                            "colors"
+                          ]
+                          ++ builtins.attrNames config.stylix
+                        )
+                      )
+                    }, but got: ${argument}")
                 )
-              }, but got: ${arg}")
-        );
-
-      # Call the configuration function with its required Stylix arguments.
-      mkConfig = fn: fn (getStylixAttrs fn);
-
-      # Safeguard configuration functions when any of their arguments is
-      # disabled.
-      mkConditionalConfig =
-        c:
-        if builtins.isFunction c then
-          let
-            allAttrsEnabled = lib.pipe c [
-              getStylixAttrs
-              builtins.attrValues
-              # If the attr has no enable option, it is instead disabled when null
-              (builtins.all (attr: attr.enable or (attr != null)))
-            ];
-          in
-          lib.mkIf allAttrsEnabled (mkConfig c)
+          );
+
+      mkConfig =
+        let
+          areArgumentsEnabled = lib.flip lib.pipe [
+            lib.attrsToList
+            (builtins.all (
+              { name, value }: value.enable or (value != null) && cfg.${name}.enable or true
```

I'd prefer ordering it like this
```suggestion
              { name, value }: cfg.${name}.enable or true && value.enable or (value != null)
```